lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -1080,6 +1080,20 @@ si32 CGHeroInstance::manaRegain() const
 	return 1 + valOfBonuses(Bonus::SECONDARY_SKILL_PREMY, 8) + valOfBonuses(Bonus::MANA_REGENERATION); //1 + Mysticism level
 }
 
+si32 CGHeroInstance::getManaTomorrow() const
```

Comment:
Here is small note why I against having any function that refer certain dates like "tomorrow".

In our bonus system any of bonuses may expire on any turn. That not require immediate fixing because vanilla Heroes 3 mechanics unlikely have anything that alter mana regen for certain time, but you must keep it in mind as anything possible in mods.

Currently I have helper called TurnInfo that currently live in pathfinder as I not yet refactored it, but it's already used in newTurn() to give hero appropriate movement points. So you may want to reuse it there for now.